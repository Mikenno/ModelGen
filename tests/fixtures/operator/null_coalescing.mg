
func f(value)
	print(value)
	return value

print(1 ?? 2)
print(1 ?? null)
print(null ?? 2)
print(null ?? null)
print()

print(f(1) ?? f(2))
print()
print(f(1) ?? f(null))
print()
print(f(null) ?? f(2))
print()
print(f(null) ?? f(null))
print()
