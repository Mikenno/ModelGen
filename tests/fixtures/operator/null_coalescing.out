1
1
2
null

1
1

1
1

null
2
2

null
null
null

